modules/standard/IO.chpl

```diff
@@ -3899,6 +3953,107 @@ proc channel.readline(ref arg: ?t): bool throws where t==string || t==bytes {
   return true;
 }
 
+/*
+  Read a line into a Chapel string. Reads until a ``\n`` is reached.
+
+  :arg arg: a string to receive the line
+  :arg maxSize: The maximum amount of characters to read.
+  :arg stripNewline: Whether to strip the trailing ``\n`` from the line.
+  :returns: `true` if a line was read without error, `false` upon EOF
+
+  :throws SystemError: Thrown if data could not be read from the channel.
```

Comment:
Please add that it will throw if the line is longer than `maxSize` and add a test verifying that it does that. If it does not, and it seems hard to change, we can make it a future and adjust it in a follow-up PR.